### PR TITLE
Add read & write timeouts to connections.

### PR DESCRIPTION
**Pull request by AlCutter**

### CODE REVIEW

File: cpp/net/connection_pool.cc [hunk 1]
```diff
@@ -32,16 +32,20 @@ using std::unique_ptr;
 using util::ClearOpenSSLErrors;
 using util::DumpOpenSSLErrorStack;
 
-DEFINE_int32(url_fetcher_max_conn_per_host_port, 4,
-             "maximum number of URL fetcher connections per host:port");
+DEFINE_int32(connection_read_timeout_seconds, 60,
+             "Connection read timeout in seconds, only applies while willing to read.");
```
Comment (phad) [inline cpp/net/connection_pool.cc]: What does 'willing to' mean here?

File: cpp/net/connection_pool.cc [hunk 2]
```diff
@@ -248,6 +252,9 @@ unique_ptr<ConnectionPool::Connection> ConnectionPool::Get(const URL& url) {
             ? base_->HttpsConnectionNew(key.first, key.second, ssl_ctx_.get())
             : base_->HttpConnectionNew(key.first, key.second),
         move(key)));
+    struct timeval read_timeout = { FLAGS_connection_read_timeout_seconds, 0 };
```
Comment (phad) [inline cpp/net/connection_pool.cc]: are the zero values millisecs?

Maybe kZeroMillis?

File: cpp/net/connection_pool.cc (hunk 1 above)
Comment (AlCutter) [inline cpp/net/connection_pool.cc]: That's yoinked from the libevent docs (which is what actually implements it):
http://www.wangafu.net/~nickm/libevent-2.0/doxygen/html/bufferevent_8h.html#a3686d744e67e9a530ebf8ab888ade2cf

Comment (AlCutter) [inline cpp/net/connection_pool.cc]: But in essence; if you're not actively trying to read or write, then it won't time out - so for example a connection idling with no requests on it wouldn't timeout.

File: cpp/net/connection_pool.cc (hunk 2 above)
Comment (AlCutter) [inline cpp/net/connection_pool.cc]: Good call - done.

**Pull request merged**